affine-cipher: Updates to the exercise instructions file (#1857)
exercism/problem-specifications: exercises/affine-cipher/description.md

the affine cipher is much stronger than the atbash cipher,
because it has many more keys.

## Encryption

The encryption function is:

  `E(x) = (ai + b) mod m`
  -  where `i` is the letter's index from `0` to the length of the alphabet - 1

>   -  `m` is the length of the alphabet. For the roman alphabet `m` is `26`.

Author: Updated.

  -  `a` and `b` are integers which make the encryption key

Values `a` and `m` must be *coprime* (or, *relatively prime*) for automatic decryption to succeed,
ie. they have number `1` as their only common factor (more information about coprime numbers

> can be found [here](https://en.wikipedia.org/wiki/Coprime_integers)). In case `a` is not coprime to `m` your

Author: I completely agree.

> not coprime to `m`, your program should indicate that this is an error. Otherwise it should

Reviewer: could change this from "program" to "code" if we want to continue the work done in https://github.com/exercism/problem-specifications/pull/528

> not coprime to `m`, your program should return `"Error: a and m must be coprime."`. Otherwise it should

Reviewer: I'm not the happiest with the word "return" here. In some languages, it is incorrect use of the type system to return a string value to signify an error. Instead, they might raise an exception, or return a None variant of their Option, or return an Left variant of their Either, or return an Err variant of their Error, or return `{:error, reason}`. For this reason, I advise against using the word "return" here. The suggestion I make would allow languages to do what is best for them.

```suggestion
not coprime to `m`, your program should indicate that this is an error. Otherwise it should
```

> program should return status `1` and `"Error: a and m must be coprime."`. Otherwise it should encrypt or

Reviewer: **optional**, since it was already this way before this PR: "return status `1`" is too specific; not all languages should signify errors this way. My suggestion will be "It is an error if `a` is not coprime to `m`."

Author: Again, I agree and have changed the text.

decrypt with the provided key.

For the purpose of this exercise, digits are valid input but they are not encrypted. Spaces and punctuation
characters are excluded. Ciphertext is written out in groups of fixed length separated by space,
the traditional group size being `5` letters. This is to make it harder to guess encrypted text based
on word boundaries.

## Decryption

The decryption function is:

  `D(y) = (a^-1)(y - b) mod m`
  -  where `y` is the numeric value of an encrypted letter, ie. `y = E(x)`
  -  it is important to note that `a^-1` is the modular multiplicative inverse (MMI)
     of `a mod m`
  -  the modular multiplicative inverse only exists if `a` and `m` are coprime.

The MMI of `a` is `x` such that the remainder after dividing `ax` by `m` is `1`:

  `ax mod m = 1`

More information regarding how to find a Modular Multiplicative Inverse

> and what it means can be found [here.](https://en.wikipedia.org/wiki/Modular_multiplicative_inverse)

Reviewer: **optional**, since it was already this way before this PR: another use of "here" as link text

Author: Updated too.

## General Examples

 - Encrypting `"test"` gives `"ybty"` with the key `a = 5`, `b = 7`
 - Decrypting `"ybty"` gives `"test"` with the key `a = 5`, `b = 7`
 - Decrypting `"ybty"` gives `"lqul"` with the wrong key `a = 11`, `b = 7`
 - Decrypting `"kqlfd jzvgy tpaet icdhm rtwly kqlon ubstx"` gives `"thequickbrownfoxjumpsoverthelazydog"` with the key `a = 19`, `b = 13`

>  - Encrypting `"test"` with the key `a = 18`, `b = 13` gives `"Error: a and m must be coprime."` because `18` and `26` are not coprime

Reviewer: ```suggestion
 - Encrypting `"test"` with the key `a = 18`, `b = 13` is an error because `18` and `26` are not coprime
```

  `E(x) = (ai + b) mod m`
  -  where `i` is the letter's index from `0` to the length of the alphabet - 1
  -  `m` is the length of the alphabet. For the Roman alphabet `m` is `26`.
  -  `a` and `b` are integers which make the encryption key

Values `a` and `m` must be *coprime* (or, *relatively prime*) for automatic decryption to succeed,

> ie. they have number `1` as their only common factor (more information can be found

Reviewer 2: ```suggestion
ie. they have number `1` as their only common factor (more information can be found in the
```

[Wikipedia article about coprime integers](https://en.wikipedia.org/wiki/Coprime_integers)). In case `a` is

> not coprime to `m` your program should return `"Error: a and m must be coprime."`. Otherwise it should

Reviewer 2: ```suggestion
not coprime to `m`, your program should return `"Error: a and m must be coprime."`. Otherwise it should
```
